### PR TITLE
Upgrade ember-cli to 3.11 

### PR DESCRIPTION
**Pull request by cadeParade**
also, runs [`es5-getter-ember-codemod`](https://github.com/ember-codemods/es5-getter-ember-codemod) which converts `this.get('foo')` into `this.foo`.

### CODE REVIEW

File: app/components/comparison-viewer-full.js [hunk 1]
```diff
@@ -15,9 +15,9 @@ export default Component.extend({
   diffImage: reads('comparison.diffImage'),
   baseImage: reads('comparison.baseScreenshot.image'),
   click() {
-    if (!this.get('comparison') || this.get('comparison.wasAdded')) {
+    if (!this.comparison || this.get('comparison.wasAdded')) {
```
Comment (unicornzero) [inline app/components/comparison-viewer-full.js]: can you replace ` this.get('comparison.wasAdded')` with ` this.comparison.wasAdded'`?

File: app/components/dashboard-nav-wrapper.js [hunk 2]
```diff
@@ -10,6 +10,6 @@ export default Component.extend({
     // hide the button if the url contains 'organizations'
     // that means we are not in the <org>/<project> namespace, we are
     // in the org settings and the 'new project' button should not show
-    return !this.get('router.currentURL').includes('organizations') && this.get('showMemberLinks');
+    return !this.get('router.currentURL').includes('organizations') && this.showMemberLinks;
```
Comment (unicornzero) [inline app/components/dashboard-nav-wrapper.js]: Can `!this.get('router.currentURL').includes('organizations')` change to `!this.router.currentURL.includes('organizations')`?

File: app/components/form-fields/checkbox-set.js [hunk 3]
```diff
@@ -18,17 +18,17 @@ export default Component.extend({
   classNames: ['FormFieldsCheckboxSet'],
 
   propertyValue: computed('changeset.isPristine', function() {
-    return this.get(`changeset.${this.get('property')}`);
+    return this.get(`changeset.${this.property}`);
```
Comment (unicornzero) [inline app/components/form-fields/checkbox-set.js]: maybe there are `get`s left in this file that can change, too?

File: app/components/form-fields/checkbox.js [hunk 4]
```diff
@@ -15,19 +15,19 @@ export default Component.extend({
   uncheckedValue: null,
 
   shouldBeChecked: computed('changeset.isPristine', 'checkedValue', function() {
-    return this.get('changeset.' + this.get('property')) === this.get('checkedValue');
+    return this.get('changeset.' + this.property) === this.checkedValue;
```
Comment (unicornzero) [inline app/components/form-fields/checkbox.js]: maybe there are gets left in this file that can change, too?

File: app/components/form-fields/input.js [hunk 5]
```diff
@@ -18,12 +18,12 @@ export default Component.extend({
   classNameBindings: ['classes'],
 
   fieldErrors: computed('changeset.error', function() {
-    return get(this.get('changeset.error'), this.get('property'));
+    return get(this.get('changeset.error'), this.property);
```
Comment (unicornzero) [inline app/components/form-fields/input.js]: maybe there are gets left in this file that can change, too?

File: app/components/form-fields/textarea.js [hunk 6]
```diff
@@ -12,6 +12,6 @@ export default Component.extend({
   classNameBindings: ['classes'],
 
   fieldErrors: computed('changeset.error', function() {
-    return get(this.get('changeset.error'), this.get('property'));
+    return get(this.get('changeset.error'), this.property);
```
Comment (unicornzero) [inline app/components/form-fields/textarea.js]: maybe there are gets left in this file that can change, too?

File: app/components/forms/slack-integration-config.js [hunk 7]
```diff
@@ -22,29 +22,29 @@ export default BaseFormComponent.extend({
 
   slackNotificationOptions: SLACK_NOTIFICATION_OPTIONS,
   selectedProject: computed('projectOptions', 'changeset.projectId', function() {
-    return selectedProjectOption(this.get('projectOptions'), this.get('changeset.projectId'));
+    return selectedProjectOption(this.projectOptions, this.get('changeset.projectId'));
```
Comment (unicornzero) [inline app/components/forms/slack-integration-config.js]: maybe there are gets left in this file that can change, too?

File: app/components/forms/usage-notification-setting.js [hunk 8]
```diff
@@ -7,19 +7,19 @@ export default BaseFormComponent.extend({
...
       });
     }
   }),
 
   actions: {
     customSave() {
-      const changeset = this.get('changeset');
+      const changeset = this.changeset;
       changeset.set('emails', changeset.get('displayEmails'));
       changeset.set('thresholds', {'snapshot-count': changeset.get('displayThresholds')});
```
Comment (unicornzero) [inline app/components/forms/usage-notification-setting.js]: maybe there are gets left in this file that can change, too?

File: app/components/organizations/integrations/integration-item.js [hunk 9]
```diff
@@ -55,41 +55,41 @@ export default Component.extend({
   }),
 
   isInstalled: computed('integrationName', function() {
-    return this.get(`organization.${this.get('integrationStatusKey')}`);
+    return this.get(`organization.${this.integrationStatusKey}`);
```
Comment (unicornzero) [inline app/components/organizations/integrations/integration-item.js]: Maybe this `this.get` can change, too?

File: app/routes/organizations/organization/setup/slack-integration.js [hunk 10]
```diff
@@ -29,12 +29,10 @@ export default Route.extend(AuthenticatedRouteMixin, {
     // This handles any other error
     if (params.error) {
       this.replaceWith('organizations.organization.integrations.slack', organization.get('slug'));
```
Comment (unicornzero) [inline app/routes/organizations/organization/setup/slack-integration.js]: Does the new syntax allow `organization.get('slug')` to change to `organization.get.slug`?

File: app/routes/organizations/organization/setup/slack-integration.js [hunk 11]
```diff
@@ -53,7 +51,7 @@ export default Route.extend(AuthenticatedRouteMixin, {
       error => {
         newIntegration.rollbackAttributes();
         this.replaceWith('organizations.organization.integrations.slack', organization.get('slug'));
```
Comment (unicornzero) [inline app/routes/organizations/organization/setup/slack-integration.js]: here too?

**Comment by cadeParade** [comment]
Thanks for reviewing! You read closely!
In some cases, we cannot replace `object.get('foo')` with `object.foo`. This is when `object` is a proxy, as in the case with `changeset`. In some cases `build` is a proxy also, due to infinity loader addon. I am not sure why `router`, `this` would be proxy objects so I'll look into those.

**Comment by cadeParade** [comment]
Check out "ES5 Getters for Computed Properties (2 of 4)" section of this page https://blog.emberjs.com/2018/04/13/ember-3-1-released.html for more info.

**Pull request merged**